src/js/service/CodeOrgMessageService.js

```diff
@@ -0,0 +1,72 @@
+/* Code.org-specific service for managing communication with the outer page
+ * when used as an embedded app in an iframe. */
+(function () {
+  var ns = $.namespace('pskl.service');
+  var MessageType = ExportedConstants.MessageType;
+
+  /**
+   * @param {!ImportService} importService
+   * @constructor
+   */
+  ns.CodeOrgMessageService = function (importService) {
+    /**
+     * The import service we'll rely on to load animations when they are
+     * selected in the outer UI.
+     * @private {!ImportService}
+     */
+    this.importService_ = importService;
+
+    /**
+     * Which origin we will receive messages from.  Set on init().
+     * @private {string}
+     */
+    this.allowedOrigin_ = null;
+  };
+
+  /**
+   * Initialize message service.  Causes service to start listening for messages
+   * on window.
+   * @param {!Window} window
+   */
+  ns.CodeOrgMessageService.prototype.init = function (window) {
+    // We only accept messages sent from the origin hosting Piskel.
+    this.allowedOrigin_ = window.location.origin;
+    window.addEventListener('message', this.receiveMessage_.bind(this));
+  };
+
+  /**
+   * Entry point for routing messages received from the Code.org parent document.
+   * @param {Event} event
+   * @private
+   */
+  ns.CodeOrgMessageService.prototype.receiveMessage_ = function (event) {
+    // Ignore messages not sent from the allowed origin
+    var origin = event.origin || event.originalEvent.origin;
+    if (origin !== this.allowedOrigin_) {
+      return;
+    }
+
+    var message = event.data;
+    // TODO: supply an exported LOAD_IMAGE constant.
+    if (message.type === MessageType.LOAD_ANIMATION) {
```

Comment:
I thought about this, and I'd like to address it once I have more message types to consider.  I was thinking that instead of exporting string constants it'd be nice to export the equivalent of Redux action-creator functions.